pkg/prometheus/promcfg.go

```diff
@@ -2223,3 +2235,76 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	}
 	return cfg
 }
+
+func (cg *ConfigGenerator) generateTracingConfig(tracingConfig *v1.PrometheusTracingConfig) (yaml.MapItem, error) {
+	cfg := yaml.MapSlice{}
+	objMeta := cg.prom.GetObjectMeta()
+
+	if tracingConfig.Endpoint == "" {
+		return yaml.MapItem{}, errors.New("Endpoint is required and can't be empty")
+	}
```

Comment:
Maybe we could add a validation marker on the API making sure this field is required(`+kubebuilder:validation:Required`), that way we could get rid of this if

Ref: https://book.kubebuilder.io/reference/markers/crd-validation.html

Reply:
With all the markers, we could probably drop a few unit tests, or replace them with a e2e test maybe 🤔

Reply:
Actually I just noticed we don't even need this annotation, since it's not tagged with +optional, it is required by default.